pyccel/codegen/printing/ccode.py

```diff
@@ -586,6 +602,68 @@ def function_signature(self, expr):
         else:
             return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_IndexedElement(self, expr):
+        if isinstance(expr.base, IndexedVariable):
+            base = expr.base.internal_variable
+        else:
+            base = expr.base
+        base = self._print(expr.base.label)
+        inds = list(expr.indices)
+        inds = inds[::-1]
+        base_shape = Shape(expr.base)
+        allow_negative_indexes = (isinstance(expr.base, IndexedVariable) and \
+                expr.base.internal_variable.allows_negative_indexes)
+        for i, ind in enumerate(inds):
+            if isinstance(ind, PyccelUnarySub) and isinstance(ind.args[0], Integer):
+                inds[i] = PyccelMinus(base_shape[i], ind.args[0])
+            else:
+                if isinstance(ind, Slice):
+                    #setting the slice start and end to their correct value if none is provided
+                    start = ind.start
+                    end = ind.end
+                    if ind.start is None:
+                        start = 0
+                    if ind.end is None:
+                        end = expr.base.shape[i]
+                    inds[i] = Slice.__new__(Slice, start, end)
+                #indices of indexedElement of len==1 shouldn't be a Tuple
+                if isinstance(ind, Tuple) and len(ind) == 1:
+                    inds[i].args = ind[0]
+                if allow_negative_indexes and not isinstance(ind, Integer):
+                    inds[i] = PyccelMod(ind, base_shape[i])
+        inds = [self._print(i) for i in inds]
+        #set dtype to the C struct types
+        dtype = self.find_in_dtype_registry(format(expr.dtype), expr.precision, array=True)
+        if expr.rank > 0:
+            return "array_slicing(%s, %s)" % (base, ", ".join(inds))
+        return "%s.nd_%s[get_index(%s, %s)]" % (base, dtype, base, ", ".join(inds))
+
+    def _print_Allocate(self, expr):
+        free_code = ''
+        #free the array if its already allocated
+        if  (expr.status != 'unallocated'):
+            free_code = 'if (%s.raw_data != NULL)\n' % self._print(expr.variable.name)
+            free_code += '{\nfree_array(%s);\n}\n' % self._print(expr.variable.name)
+        self._additional_imports.add('ndarrays')
+        shape = ", ".join(str(a) for a in expr.shape)
+        dtype = self.find_in_dtype_registry(format(expr.variable.dtype), expr.variable.precision, array=True)
+        shape_name = "shape_dummy_%s" % format(expr.__hash__()%991)
+        init_shape = "int %s[] = {%s};" % (shape_name, shape)
+        alloc_code = "{} = array_create({}, {}, nd_{});".format(expr.variable, len(expr.shape), shape_name, dtype)
+        return '{}{}\n{}'.format(free_code, init_shape, alloc_code)
+
+    def _print_Slice(self, expr):
+        # print(expr.args)
+        if expr.start is None or  isinstance(expr.start, Nil):
+            start = '-1'
```

Comment:
Why not 0?

Reply:
i was passing -1 to check if the either the start or end are none (0 won't give me that ability) i did manage to handles it in line 620 this won't be needed anymore

Reply:
Why do you need that ability?

Reply:
i was going to manage that in the C code but i am managing it from `_print_IndexedElement` now.
i just forget to remove the ifs

Reply:
I was just wondering what there is to manage. If you pass `None` to `start` then doesn't it default to `0`?

Reply:
ah, i just wanted to manage them in the same place